gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/util/EhrResourceExtractorUtil.java

```diff
@@ -37,6 +38,30 @@ private static boolean validPlanStatement(RCMRMT030101UK04Component4 component,
         return component.getPlanStatement() != null && component.getPlanStatement().getId() == resourceId;
     }
 
+    public static List<RCMRMT030101UK04EhrComposition> extractValidImmunizationEhrCompositions(RCMRMT030101UK04EhrExtract ehrExtract) {
```

Comment:
You could move this public method higher, so all the public methods are above the private methods.